design/legacy-design/device-service/discovery.md

```diff
@@ -0,0 +1,79 @@
+## Dynamic Device Discovery
```

Comment:
In my view, I'm not sure there is full agreement on a dynamic device discovery approach.  @tsconn23 had a separate proposal and I'm not sure whether or not both of these approaches have been fully discussed or reconciled.  I also think the approach taken may be impacted by the current open discussion on V2 API implementation.  

I'd ask that this PR be broken up and a separate PR be created for each of the documents being added to the repository.

Reply:
I think the design here isn't really affected by Trevor's proposed sequences for device creation and potentially migration. Where we get to the actual device creation that may have to change in the light of the v2 API but that applies everywhere we create.

I've split the PR as noted below.

Reply:
Added @tsconn23 as reviewer.

Reply:
During discussion in Device WG 20-Jan-20 we confirmed that this proposal targets V1 implementation for Geneva. I think in the current state, I'm fine with the proposal being merged as an initial draft which can be refined subsequently.